agent/structs/structs.go

```diff
@@ -1812,6 +1820,12 @@ type IndexedCheckServiceNodes struct {
 	QueryMeta
 }
 
+type IndexedNodesWithGateways struct {
```

Comment:
What do you think about naming it something more general, i.e. `IndexedServiceDump` or `IndexedDump` or something? Not quite sure what a better name would be though. That way we if we ever need to add more fields to it we do not have to change the name again.

Reply:
Sounds good to me. Done.

Reply:
Ah, remembered that there's already an `IndexedServiceDump` for the `ServiceDump` type. Reverting. If we need to add more I think we can cross the renaming bridge when we get there.